starlette_web/contrib/scheduler/backends/posix.py

```diff
@@ -0,0 +1,75 @@
+import copy
+import logging
+import os
+import tempfile
+
+from starlette_web.contrib.scheduler.backends.base import BasePeriodicTaskScheduler
+from starlette_web.contrib.scheduler.app_settings import PosixSettings
+
+
+logger = logging.getLogger("starlette_web.contrib.scheduler")
+
+
+class CrontabScheduler(BasePeriodicTaskScheduler):
+    settings_class = PosixSettings
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.crontab_lines = []
+
+    def _read_jobs(self):
+        self.crontab_lines = os.popen("%s -l" % self.settings.CRONTAB_EXECUTABLE).readlines()
+
+    def _write_jobs(self):
+        fd, path = tempfile.mkstemp()
+        tmp = os.fdopen(fd, "w")
+        for line in self.crontab_lines:
+            if not line.strip():
+                continue
+            tmp.write(line.strip() + "\n")
+        tmp.close()
+        os.system("%s %s" % (self.settings.CRONTAB_EXECUTABLE, path))
```

Comment:
правильно я тебя понял, что через наш кронтаб можно будет запустить что угодно (ну то есть то, что можно запустить через CRONTAB_EXECUTABLE?)
есть пара вопросов:
1) от имени какого юзера будет выполняться ?
2) на сколько этот подход безопасен (пока не придумал вектор атаки, но есть переживания в этом плане)
3) можем не использовать старый стиль форматирования?
```suggestion
        os.system(f"{self.settings.CRONTAB_EXECUTABLE} {path}")
```

Reply:
Не совсем, кронтаб всегда запускает команду

```
cd PROJECT_ROOT && sys.executable command.py scheduler --cmd=run --jobhash=...
```

Реализация для крона вообще почти 1-в-1 взята отсюда https://github.com/kraiz/django-crontab/tree/master/django_crontab
А для винды -- адаптирована под django_crontab

Соответственно,
1) добавление записей в кронтаб производится под виртуальным окружением проекта и от того же пользователя, у кого развернут проект и выполняться будет от того же пользователя / на винде я сделал опциональные параметры username / password, т.к. может не быть админских прав для добавления таски, а по умолчанию от SYSTEM

2) зависит от команды, которую юзер напишет, имхо

3) поправлю позже, просто я пару раз садился за переписывание django-crontab, потом психовал и откатывал как у них, поэтому под конец уже ничего не стал менять